### PR TITLE
[13.0][ADD] helpdesk_mgmt_crm: New addon.

### PR DESCRIPTION
**Pull request by victoralmau**
New addon.

Please @pedrobaeza and @CarlosRoca13 can you review it?

@Tecnativa TT38675

### CODE REVIEW

File: helpdesk_mgmt_crm/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Copyright 2022 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+{
+    "name": "Helpdesk Management - Crm",
```
Comment (pedrobaeza) [inline helpdesk_mgmt_crm/__manifest__.py]:
```suggestion
    "name": "Link between Helpdesk and CRM",
```

File: helpdesk_mgmt_crm/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    "summary": "Links helpdesk tickets with leads",
+    "version": "13.0.1.0.1",
```
Comment (pedrobaeza) [inline helpdesk_mgmt_crm/__manifest__.py]:
```suggestion
    "version": "13.0.1.0.0",
```

File: helpdesk_mgmt_crm/README.rst [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+.. |badge4| image:: https://img.shields.io/badge/weblate-Translate%20me-F47D42.png
+    :target: https://translation.odoo-community.org/projects/helpdesk-13-0/helpdesk-13-0-helpdesk_mgmt_crm
+    :alt: Translate me on Weblate
+.. |badge5| image:: https://img.shields.io/badge/runbot-Try%20me-875A7B.png
+    :target: https://runbot.odoo-community.org/runbot/282/13.0
+    :alt: Try me on Runbot
+
+|badge1| |badge2| |badge3| |badge4| |badge5| 
+
+This module allows you to create (and transfer messages) an opportunity from a helpdesk ticket, and keeps it linked.
```
Comment (pedrobaeza) [inline helpdesk_mgmt_crm/README.rst]:
```suggestion
This module allows you to convert a helpdesk ticket into an opportunity, transferring its messages,
and keeping both linked.
```

Then add a USAGE.rst file for explaining how to do the conversion.

File: helpdesk_mgmt_crm/models/crm_lead.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# Copyright 2022 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class CrmLead(models.Model):
+    _inherit = "crm.lead"
+
+    ticket_id = fields.Many2one(comodel_name="helpdesk.ticket", string="Ticket",)
```
Comment (pedrobaeza) [inline helpdesk_mgmt_crm/models/crm_lead.py]:
```suggestion
    ticket_id = fields.Many2one(comodel_name="helpdesk.ticket", string="Ticket")
```

File: helpdesk_mgmt_crm/models/helpdesk_ticket.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+import html2text
+
+from odoo import SUPERUSER_ID, _, fields, models
+from odoo.exceptions import UserError
+
+
+class HelpdeskTicket(models.Model):
+    _inherit = "helpdesk.ticket"
+
+    lead_id = fields.Many2one(comodel_name="crm.lead", string="Opportunity",)
```
Comment (pedrobaeza) [inline helpdesk_mgmt_crm/models/helpdesk_ticket.py]:
```suggestion
    lead_id = fields.Many2one(comodel_name="crm.lead", string="Opportunity")
```

**Review by pedrobaeza — COMMENTED** [review]
I think we need to add a wizard to select the sales team and/or the salesman, to get benefit of the automatic notifications provided on both of them (when you have the subtype "Opportunity created" in the sales team).

**Comment by victoralmau** [comment]
Changes done and wizard added.

File: helpdesk_mgmt_crm/views/helpdesk_ticket_view.xml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+            eval="[(6, 0, [ref('sales_team.group_sale_salesman')])]"
+        />
+        <field name="arch" type="xml">
+            <xpath expr="//header" position="inside">
+                <button
+                    class="oe_highlight"
+                    type="action"
+                    name="%(helpdesk_ticket_create_lead_action)d"
+                    icon="fa-arrow-right"
+                    string="Create Lead"
```
Comment (pedrobaeza) [inline helpdesk_mgmt_crm/views/helpdesk_ticket_view.xml]:
```suggestion
                    string="Convert to opportunity"
```

File: helpdesk_mgmt_crm/wizard/helpdesk_ticket_create_lead.py [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        return vals
+
+    def _prepare_vals(self):
+        return {
+            "ticket_id": self.ticket_id.id,
+            "name": self.ticket_id.name,
+            "partner_id": self.ticket_id.partner_id.id,
+            "user_id": self.user_id.id or self.ticket_id.user_id.id,
+            "team_id": self.team_id.id,
+            "description": html2text.html2text(self.ticket_id.description),
```
Comment (pedrobaeza) [inline helpdesk_mgmt_crm/wizard/helpdesk_ticket_create_lead.py:36]:
```suggestion
            "description": html2text.html2text(self.ticket_id.description),
            "type": "opportunity",
```

File: helpdesk_mgmt_crm/wizard/helpdesk_ticket_create_lead_views.xml [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+                <group>
+                    <field name="ticket_id" />
+                    <field name="user_id" />
+                    <field name="team_id" />
+                </group>
+                <footer>
+                    <button
+                        type="object"
+                        name="action_helpdesk_ticket_to_lead"
+                        string="Create Lead"
```
Comment (pedrobaeza) [inline helpdesk_mgmt_crm/wizard/helpdesk_ticket_create_lead_views.xml]:
```suggestion
                        string="Create opportunity"
```

File: helpdesk_mgmt_crm/views/helpdesk_ticket_view.xml [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+                    attrs="{'invisible' : [('lead_id', '!=', False)]}"
+                />
+            </xpath>
+            <xpath expr="//div[@name='button_box']" position="inside">
+                <button
+                    class="oe_stat_button"
+                    icon="fa-star"
+                    type="object"
+                    name="action_open_lead"
```
Comment (pedrobaeza) [inline helpdesk_mgmt_crm/views/helpdesk_ticket_view.xml]: No smart-button with the count and saying "Opportunity(ies)"?

File: helpdesk_mgmt_crm/wizard/helpdesk_ticket_create_lead.py [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+        }
+
+    def action_helpdesk_ticket_to_lead(self):
+        lead = self.env["crm.lead"].create(self._prepare_vals())
+        self.ticket_id.write({"lead_id": lead.id})
+        self.ticket_id.message_ids.write({"model": lead._name, "res_id": lead.id})
+        # Chatter reflects new Lead
+        body = _(
+            "This ticket has been converted to "
```
Comment (pedrobaeza) [inline helpdesk_mgmt_crm/wizard/helpdesk_ticket_create_lead.py]:
```suggestion
            "This ticket has been converted to the opportunity "
```

File: helpdesk_mgmt_crm/readme/USAGE.rst [hunk 11]
```diff
@@ -0,0 +1,7 @@
+#. Go to *Helpdesk > Tickets* and create a new one or enter one.
+#. Click on "Create lead" button.
+#. A wizard will be displayed to set user and sales team if you want.
+#. Click on "Create lead" button.
+#. A lead will have been created with the ticket data and the messages will have been moved.
+#. The related ticket will be displayed so you can return to it.
+#. A "Opportunity" smart-button is also displayed on the ticket.
```
Comment (pedrobaeza) [inline helpdesk_mgmt_crm/readme/USAGE.rst]:
```suggestion
#. An "Opportunity" smart-button is also displayed on the ticket.
```

File: helpdesk_mgmt_crm/models/crm_lead.py [hunk 12]
```diff
@@ -0,0 +1,10 @@
+# Copyright 2022 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class CrmLead(models.Model):
+    _inherit = "crm.lead"
+
+    ticket_id = fields.Many2one(comodel_name="helpdesk.ticket", string="Ticket")
```
Comment (pedrobaeza) [inline helpdesk_mgmt_crm/models/crm_lead.py]:
```suggestion
    ticket_id = fields.Many2one(comodel_name="helpdesk.ticket", string="Helpdesk Ticket")
```

File: helpdesk_mgmt_crm/models/helpdesk_ticket.py [hunk 13]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2022 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, fields, models
+
+
+class HelpdeskTicket(models.Model):
+    _inherit = "helpdesk.ticket"
+
+    lead_id = fields.Many2one(comodel_name="crm.lead", string="Opportunity")
```
Comment (pedrobaeza) [inline helpdesk_mgmt_crm/models/helpdesk_ticket.py]: This should be a one2many.

Comment (victoralmau) [reply helpdesk_mgmt_crm/models/helpdesk_ticket.py]: Sorry, I thought this was a many2one because a ticket can only be converted into 1 lead, right?

Comment (pedrobaeza) [reply helpdesk_mgmt_crm/models/helpdesk_ticket.py]: I wouldn't restrict it. Imagine a customer asks for 3 products at the same time and you want to redirect each product to one salesman...

**Comment by elvise** [comment]
@victoralmau i’m testing your PR, what happen if one ticket create 2 or more opportunities ?
I think it’s better 1 Ticket - N Opportunities 

What do you think ?

**Comment by pedrobaeza** [comment]
See https://github.com/OCA/helpdesk/pull/358#discussion_r966208422 :wink:

**Comment by elvise** [comment]
> See [#358 (comment)](https://github.com/OCA/helpdesk/pull/358#discussion_r966208422) 😉

Oh Sorry 😅

File: helpdesk_mgmt_crm/models/helpdesk_ticket.py (hunk 13 above)
Comment (pedrobaeza) [reply helpdesk_mgmt_crm/models/helpdesk_ticket.py]: For supporting this case, let's copy the messages instead of transferring them.

**Comment by victoralmau** [comment]
Changes done and one2many field added.

**Review by pedrobaeza — APPROVED** [review]

**Review by elvise — CHANGES_REQUESTED** [review]
From the user point of view i guess it’s better “Create Opportunity” instead “Convert in Opportunity”

**Comment by pedrobaeza** [comment]
The text is "Convert to opportunity", and it follows the same nomenclature as leads converted to opportunity and the usual salesman pipeline.

**Comment by elvise** [comment]
> The text is "Convert to opportunity", and it follows the same nomenclature as leads converted to opportunity and the usual salesman pipeline.

@pedrobaeza From this perspective looks reasonable

**Review by elvise — APPROVED** [review]
LGTM!

**Comment by francescapenso** [comment]
This will be ported in 14.0 and 16?

**Comment by victoralmau** [comment]
No, this module is only in v13 yet. You can migrate to higher versions if you wish.

**Pull request merged**